lib/Controller/RecipeController.php

```diff
@@ -100,7 +101,16 @@ public function update($id) {
 		$this->dbCacheService->triggerCheck();
 		
 		$recipeData = $this->restParser->getParameters();
-		$file = $this->service->addRecipe($recipeData);
+		try {
+			$file = $this->service->addRecipe($recipeData);
+		} catch (NoRecipeNameGivenException $ex) {
+			$json = [
+				'msg' => $ex->getMessage(),
+				'file' => $ex->getFile(),
+				'line' => $ex->getLine(),
+			];
+			return new JSONResponse($json, Http::STATUS_NOT_ACCEPTABLE);
```

Comment:
Is this really the correct status code to return here? Isn’t this more like a 400?

Reply:
Yes, you are right. 400 seems more suited. I do not like the 400 as it is items used as a catch all code. But we do not use it currently as far as I know. 

I will update soon.

Reply:
I fixed this.

Reply:
I understood your intention, but looking through the 4xx error codes on https://developer.mozilla.org/en-US/docs/Web/HTTP/Status only 422 seemed suited but is restricted to WebDAV. we could misuse it though to have a more specific code...

Reply:
Would you prefer the 400 or the 422? I'd maybe even go for the 422 as we are building the API in any case. So I would maybe even vote for 422. I will wait with merging for your thoughts.

We can of course say that for that specific endpoint (`/api/recipe/<id>/image`) the 400 is sufficient and we provide an error message as well...

Reply:
I’d support either way, take your pick ;)